src/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProvider.php

```diff
@@ -447,9 +455,33 @@ private function extractOptionsData(ProductForEditing $productForEditing): array
             ],
             'customizations' => $this->extractCustomizationsData($productForEditing),
             'suppliers' => $this->extractSuppliersData($productForEditing),
+            'attachments' => $this->extractAttachmentsData($productForEditing),
         ];
     }
 
+    /**
+     * @param ProductForEditing $productForEditing
+     *
+     * @return array<string, array<int, array<string, mixed>>>
+     */
+    private function extractAttachmentsData(ProductForEditing $productForEditing): array
```

Comment:
Could you add unit test ?

Reply:
Additional request from @eternoendless : could we use https://phpstan.org/writing-php-code/phpdoc-types#array-shapes to acknowledge the output structure instead of `array<string, array<int, array<string, mixed>>>`

Reply:
Do we really want to mix it up now ? We've been using psalm for quite a while, I don't think those shapes gives that much of a value.
If yes, then shouldn't we just try make a dedicated PR to change them everywhere?

Reply:
I think the array shape structure gives more info for hashmaps arrays (you can see the keys used which is really a bonus) Of course the idea is not to change everything in this PR but at least start to use it when it's relevant
Later we can make a dedicated PR to improve them all